### PR TITLE
[Dell] z9100 port qos & buffer changes of 0330 to master

### PR DESCRIPTION
**Pull request by vharish02**
DO NOT MERGE. Please merge after or along with PR 2239
https://github.com/Azure/sonic-buildimage/pull/2239

This commit has the forward porting of changes from 20180330
to master. Unit tested by checking the broadcom registers
for the values. The values in hardware reflect the correct values
except for TC to PFC priority group map values. Master branch values
for TC to PFC priority group map for both z9100 and s6100 in incorrect,
I have a build of Oct 12 master and the values are correct in
that build.

Signed-off-by: Harish Venkatraman <Harish_Venkatraman@Dell.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
retest this please

File: device/dell/x86_64-dell_z9100_c2538-r0/minigraph.xml [hunk 1]
```diff
@@ -0,0 +1,1079 @@
+<DeviceMiniGraph xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns="Microsoft.Search.Autopilot.Evolution">
```
Comment (lguohan) [inline device/dell/x86_64-dell_z9100_c2538-r0/minigraph.xml]: remove minigraph, use default_sku

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-C8D48/qos.json.j2 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{%- macro generate_port_lists(PORT_ALL) %}
+    {# Generate list of ports #}
+    {% for port_idx in range(0, 128, 2) if port_idx not in [26, 30, 34, 38, 106, 110, 114, 118] %}
+        {% if PORT_ALL.append("Ethernet%d" % (port_idx)) %}{% endif %}
+    {% endfor %}
+{%- endmacro %}
```
Comment (wendani) [inline device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-C8D48/qos.json.j2]: generate_port_lists() will not be called

**Comment by vharish02** [comment]
> retest this please

yes will test it

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-C8D48/qos.json.j2 (hunk 2 above)
Comment (vharish02) [reply device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-C8D48/qos.json.j2]: will remove

File: device/dell/x86_64-dell_z9100_c2538-r0/minigraph.xml (hunk 1 above)
Comment (vharish02) [reply device/dell/x86_64-dell_z9100_c2538-r0/minigraph.xml]: ok

**Comment by lguohan** [comment]
can you rebase to master, seems build is failing.

**Comment by vharish02** [comment]
> > can you rebase to master, seems build is failing.
> 
> Yes will do.

Guohan the rebase was done and builds were successful, can you merge this PR.

**Pull request merged**